scripts/compile_tests/failures_histogram.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import re
+
+from common import get_testcases, open_test_results, skipped_test
+
+
+"""
+python failures_histogram.py dynamo_logs_for_py311/
+
+Analyzes skip reasons for Dynamo tests and prints a histogram with repro
+commands. You'll need to download the test reports for the Dynamo shards
+and put them under the specified directory.
+"""
+
+
+def skip_reason(testcase):
+    for child in testcase.iter():
+        if child.tag != "skipped":
+            continue
+        return child.attrib["message"]
+    raise AssertionError("no message?")
+
+
+IGNORED_REASONS = {
```

Comment:
Should we mention why we skip some of these? Like e.g. why is "Skipped" or "Skipped!" skipped?